wp-includes/rest-api/endpoints/class-wp-rest-customize-partials-controller.php

```diff
@@ -0,0 +1,327 @@
+
+		$schema = array(
+			'$schema'    => 'http://json-schema.org/schema#',
+			'title'      => 'partial',
+			'type'       => 'object',
+			'properties' => array(
+				'fallback_refresh' => array(
+					'description' => __( 'Whether to refresh the entire preview in case a partial cannot be refreshed.' ),
+					'type'        => 'boolean',
+					'context'     => array( 'embed', 'view', 'edit' ),
+					'readonly'    => true,
+				),
+				'id'              => array(
+					'description' => __( 'Identifier for the partial.' ),
+					'type'        => 'string',
+					'context'     => array( 'embed', 'view' ),
+					'readonly'    => true,
+				),
+				'container_inclusive' => array(
+					'description' => __( 'Whether the container element is included in the partial' ),
+					'type'        => 'boolean',
+					'context'     => array( 'embed', 'view' ),
+					'readonly'    => true,
+				),
+				'selector'        => array(
+					'description' => __( 'The jQuery selector to find the container element for the partial.' ),
+					'type'        => 'string',
+					'context'     => array( 'embed', 'view', 'edit' ),
+					'readonly'    => true,
+				),
+				'settings'        => array(
+					'description' => __( 'IDs for settings tied to the partial.' ),
+					'type'        => 'array',
+					'context'     => array( 'embed', 'view', 'edit' ),
+					'readonly'    => true,
+				),
+				'type'            => array(
+					'description' => __( 'Type of the partial.' ),
+					'type'        => 'string',
+					'context'     => array( 'view' ),
+					'readonly'    => true,
+				),
+			),
+		);
+
+		return $this->add_additional_fields_schema( $schema );
+	}
+
+	/**
+	 * Checks if a given request has access to read a partial.
+	 *
+	 * @since 4.?.?
+	 * @access public
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return bool|WP_Error True if the request has read access for the item, WP_Error object otherwise.
+	 */
+	public function get_item_permissions_check( $request ) {
+		$wp_customize = $this->ensure_customize_manager();
+
+		$wp_customize->selective_refresh->add_dynamic_partials( array( $request['partial'] ) );
+		$partial = $wp_customize->selective_refresh->get_partial( $request['partial'] );
+		if ( ! $partial ) {
+			return new WP_Error( 'rest_partial_invalid_id', __( 'Invalid partial ID.' ), array(
+				'status' => 403,
+			) );
+		}
+
+		return $partial->check_capabilities();
+	}
+
+	/**
+	 * Retrieves a single partial.
+	 *
+	 * @since 4.?.?
+	 * @access public
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function get_item( $request ) {
+		$wp_customize = $this->ensure_customize_manager();
+
+		$wp_customize->selective_refresh->add_dynamic_partials( array( $request['partial'] ) );
+		$partial = $wp_customize->selective_refresh->get_partial( $request['partial'] );
```

Comment:
I suppose it will not happen in the end because `get_item_permissions_check` will short-circuit the check, but good to add anyway for completeness.